internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -593,7 +593,7 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				Type:     pluginsdk.TypeBool,
 				Optional: true,
 				Default:  false,
-				ForceNew: true,
+				ForceNew: false,
```

Comment:
It is better for remove the `ForceNew` annotation here at all
```suggestion
```

Reply:
Thanks @aristosvo, I have added the change per your comments in https://github.com/hashicorp/terraform-provider-azurerm/pull/13413/commits/8c29589dee193c3c36ee7708d0fc56ff433e17dc